ex04.ex

```diff
@@ -75,7 +92,38 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+  def even_odd([]) do
```

Comment:
# 6

Two problems here:

First, you scan the list twice to build the two output lists. You can do it in a single scan.

Second, there's an awful lot of duplication. You can use functions to remove it. Passing functions to other functions is almost like subclassing in OO languages.